src/shared/util.js

```diff
@@ -793,6 +793,11 @@ function stringToPDFString(str) {
   return strBuf.join("");
 }
 
+function escapeString(str) {
```

Comment:
> Also, if the string contains unicode we need to ensure the string starts with the unicode BOM.

It tryed to add an UTF-16BE BOM but the string is still considered as normal string, for example `<FEFF0041>` is equivalent to `"\xfe\xff\00\x41"` when used with command `Tj`.
I tryed this pdf in pdf.js:
```
%PDF-1.4
1 0 obj <</Type /Catalog /Pages 2 0 R>>
endobj
2 0 obj <</Type /Pages /Kids [3 0 R] /Count 1>>
endobj
3 0 obj<</Type /Page /Parent 2 0 R /Resources 4 0 R /MediaBox [0 0 500 800] /Contents 6 0 R>>
endobj
4 0 obj<</Font <</F1 5 0 R>>>>
endobj
5 0 obj<</Type /Font /Subtype /Type1 /BaseFont /Helvetica>>
endobj
6 0 obj
<</Length 44>>
stream
BT /F1 24 Tf 175 720 Td <FEFF004100420043> Tj ET
endstream
endobj
xref
0 7
0000000000 65535 f
0000000009 00000 n
0000000056 00000 n
0000000111 00000 n
0000000212 00000 n
0000000250 00000 n
0000000317 00000 n
trailer <</Size 7/Root 1 0 R>>
startxref
413
%%EOF
```
and the string is displayed as a 8 chars string.
So I guess that when we've a non-ascii string we must find a font which can display it and add the necessary resources and then re-encode the string for this font.
We could probably cheat when printing in trying to render the string on the canvas with an unicode font but for the saving part we need to handle that correctly.
So if I'm correct we could probably do that later in an other patch.
Wdyt ?

Reply:
I agree that that should be done in a follow-up, mainly to limit the scope of this patch but also because it's not a big problem for now (indeed, for saving this should be taken care of and can then also be done for rendering).